### ISSUE TITLE
Multithreaded builds (/m)

### ISSUE THREAD

**Issue by Morten242** [Microsoft/vcpkg#258]
When building the switch to enabled multi-threaded builds is not set. Is there a reason why it can't be done/isn't already enabled? Should we? Should/can it be enabled for every port? It's an easy way to decrease build-times for a lot of projects.

**Comment by ras0219-msft** [comment]
We should definitely use this switch in `vcpkg_build_msbuild`.

We already pass this switch in `vcpkg_build_cmake`: https://github.com/Microsoft/vcpkg/blob/master/scripts/cmake/vcpkg_build_cmake.cmake#L4.

Additionally, we should look at using Ninja as our default buildsystem target for CMake since it can parallelize even further than MSBuild.

**Comment by Morten242** [comment]
Ah, that we do. However, vcpkg_build_cmake only seems to be used in a few ports:

```powershell
» ls portfile.cmake -r | sls vcpkg_build_cmake

bullet3\portfile.cmake:26:vcpkg_build_cmake()
geos\portfile.cmake:31:vcpkg_build_cmake()
glbinding\portfile.cmake:19:#vcpkg_build_cmake()
libodb\portfile.cmake:25:vcpkg_build_cmake()
libodb-sqlite\portfile.cmake:25:vcpkg_build_cmake()
sqlite3\portfile.cmake:18:vcpkg_build_cmake()
```

While vcpkg_install_cmake (which also builds, if not already built) is used in a few more:

```powershell
» ls portfile.cmake -r | sls vcpkg_install_cmake | measure | select count

Count
-----
   58
```

So this would mean we have a bug with most portfiles in that they don't perform vcpkg_cmake_build before install.

Now, if we could add the multithreaded-build switch to vcpkg_cmake_install we could essentially remove the build script as they'd now be the same (except for install target). Or we could add the ``vcpkg_cmake_build()`` line to all ports that are missing it.

**Comment by ras0219-msft** [comment]
We should definitely add /m to the install step, since that's obviously a pure win.

As stated in the PR, I don't think adding the build step back into all the existing portfiles adds much value -- they clearly work currently, so we're trading a marginal debugging benefit for the risk of (much) slower builds in the case of libraries with improper dependency management. Thus, I think new portfiles can decide to have it or not as they see most fit.

**Comment by ras0219-msft** [comment]
Fixed by #273 and #274.
